traffic_ops/testing/api/v4/cachegroups_parameters_test.go

```diff
@@ -79,6 +90,114 @@ func CreateTestCacheGroupParameters(t *testing.T) {
 	testData.CacheGroupParameterRequests = append(testData.CacheGroupParameterRequests, resp.Response...)
 }
 
+func CreateTestCacheGroupParametersAlreadyExist(t *testing.T) {
+
+	resp, _, err := TOSession.GetAllCacheGroupParameters(client.RequestOptions{})
+	cachegroupParameters := resp.Response
+	getAllcgparameters := cachegroupParameters.CacheGroupParameters
+	parameterID := *getAllcgparameters[0].Parameter
```

Comment:
There are two possible segfaults on this line, `*` and `[0]`

Reply:
added null checks.

Reply:
This can still segfault if the response did not contain a `parameter` property at index `0`